### ISSUE TITLE
Lexical error

### ISSUE THREAD

**Issue by tronicek** [javaparser/javaparser#2269]
When parsing file jaxws\src\share\jaxws_classes\com\sun\xml\internal\rngom\parse\compact\CompactSyntax.java from OpenJDK, it throws the following exception:

Exception in thread "main" com.github.javaparser.ParseProblemException: Lexical error at line 244, column 38.  Encountered: "n" (110), after : "\'\\u005c"

Line 244 is as follows:
s = s.replace('\u0000', '\u005cn');

**Comment by matozoid** [comment]
That currently requires the unicode preprocessor to be turned on. --> https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/ParserConfiguration.java#L291

The grammar could be changed to support this by default, so let's leave this open.

**Comment by paulk-asert** [comment]
The following file also seems to display this error:
https://github.com/apache/groovy/blob/master/src/main/java/groovy/transform/builder/DefaultStrategy.java
Error: Lexical error at line 287, column 0.  Encountered: <EOF> after : ""
(but as noted for the other example, I'd guess that line number is somewhat after the problematic code)

**Comment by jlerbsc** [comment]
Hi @tronicek have you try this turnaround (ParserConfiguration.setPreprocessUnicodeEscapes(..)?

**Comment by tronicek** [comment]
Yes. It works for me. I currently do not have any problem with JavaParser.

**Comment by jlerbsc** [comment]
@tronicek Very well could you close this issue please?

**Comment by tronicek** [comment]
See @matozoid's comment: "The grammar could be changed to support this by default, so let's leave this open." If @matozoid changed his mind, he can close it by himself.
